Assistant: enable positron assistant git integration by default

sharon-wang opened the pull request:
### Summary

- addresses https://github.com/posit-dev/positron/issues/9066
- sets the default for `positron.assistant.gitIntegration.enable` to true and removes the experimental tag, but keeps the flag for now
- disables the `Source Control...` context item on Web due to https://github.com/posit-dev/positron/issues/9181

### Release Notes

#### New Features

- Assistant: git integration Assistant tools available by default (#9066)

### QA Notes

See this PR for usage: https://github.com/posit-dev/positron/pull/8257

#### Git features enabled by default!

- `positron.assistant.gitIntegration.enable` setting still available in the settings UI but is now enabled by default

#### `Source Control...` context option disabled by default for Web/Workbench

- On Web / Workbench, the `Source Control...` context option should be missing:

    <img width="672" height="211" alt="image" src="https://github.com/user-attachments/assets/b3d9202c-0e55-4ea4-906c-71b327d7367a" />

- On Desktop, it should be available (some extra options because of my clipboard and editor state):

    <img width="607" height="279" alt="image" src="https://github.com/user-attachments/assets/2103b951-14f0-4c31-9eae-ddc1caeda48f" />

#### `positron.assistant.gitIntegration.enable` setting can be used to make `Source Control...` context option available on Web/Workbench

- Add `"positron.assistant.sourceControlHistoryContext.enable": true` to the settings JSON in a web build to enable the `Source Control...` context option (helpful for debugging)

sharon-wang committed:
Assistant: enable positron assistant git integration by default

georgestagg reviewed (dismissed):
https://github.com/posit-dev/positron/pull/9075 has been merged, fixing a bug with git integration.

With that done, this LGTM!

jonvanausdeln commented:
Might want to wait for a decision about https://github.com/posit-dev/positron/issues/9125 before merging this.

sharon-wang committed:
disable SCM History Chat Context in Positron Web due to path error

see https://github.com/posit-dev/positron/issues/9181

sharon-wang commented:
An update!

- https://github.com/posit-dev/positron/issues/9125 will be addressed in https://github.com/posit-dev/positron/pull/9178 <-- will be merged before this PR
- this PR enables git integration by default on Positron Desktop and Workbench
- this PR disables the `Source Control...` context option for Positron Web/Workbench due to https://github.com/posit-dev/positron/issues/9181 which is slated for Triage

sharon-wang committed:
add setting to enable source control context in Web

this will be helpful for enabling the source control context to debug the error in a web build, but still disables it by default for web and enables it by default for desktop
